### ISSUE TITLE
Bug: no-unused-private-class-members ignores AccessorProperty

### ISSUE THREAD

**Issue by Kuldeep2822k** [eslint/eslint#20887]
### Environment

Node version: v24.15.0
npm version: latest
Local ESLint version: v10.4.0
Global ESLint version: none
Operating System: Windows

### What parser are you using?

@typescript-eslint/parser

### What did you do?

```js
const tsParser = require("@typescript-eslint/parser");

module.exports = [
  {
    languageOptions: {
      parser: tsParser,
      ecmaVersion: "latest",
      sourceType: "module"
    },
    rules: {
      "no-unused-private-class-members": "error"
    }
  }
];
```
</details>

```js
class A {
  accessor #x; // Unused private auto-accessor
}
```

### What did you expect to happen?

The `no-unused-private-class-members` rule should report an error because `#x` is defined but never used within the class.

### What actually happened?

No error is reported. The rule completely ignores the unused member.

### Link to Minimal Reproducible Example
https://eslint.org/play/#eyJ0ZXh0IjoiY2xhc3MgQSB7XG4gIGFjY2Vzc29yICN4OyAvLyBVbnVzZWQgcHJpdmF0ZSBhdXRvLWFjY2Vzc29yXG59XG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLXVudXNlZC1wcml2YXRlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9fSwicGFyc2VyIjoiQHR5cGVzY3JpcHQtZXNsaW50L3BhcnNlciJ9fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [ ] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

The rule's AST visitor currently listens for `PropertyDefinition` and `MethodDefinition` nodes, but it completely skips `AccessorProperty` nodes (Stage 3 auto-accessors).

**Comment by mdjermanovic** [comment]
> `AccessorProperty` nodes (Stage 3 auto-accessors)

Per our [policies](https://github.com/eslint/eslint#what-about-experimental-features), core rules support only stage 4 language features. If and when the proposal for this syntax reaches stage 4, we'll add support for it.
